spec/functions/delete_spec.rb

```diff
@@ -4,19 +4,22 @@
   it { is_expected.not_to eq(nil) }
   it { is_expected.to run.with_params().and_raise_error(Puppet::ParseError) }
   it { is_expected.to run.with_params([]).and_raise_error(Puppet::ParseError) }
-  it { is_expected.to run.with_params([], 'two', 'three').and_raise_error(Puppet::ParseError) }
+  it { is_expected.to run.with_params([], 'two', 'three') }
```

Comment:
this first block of tests test that the wrong number of arguments raises an error. as you're adding an optional third param, the correct fix here is 

it { is_expected.to run.with_params([], 'two', 'three', 'four').and_raise_error(Puppet::ParseError) }